lib/epochs.js

```diff
@@ -286,6 +295,93 @@ module.exports = function Epochs(ssb) {
     })
   }
 
+  /* private helpers */
+
+  function epochsReduce(groupId, opts = {}, cb) {
+    const { getters } = opts
+    // - `groupId` *String* ssb-uri for a group
+    // - `getters` *Object* which describes which data fields you would like
+    //   added to each epoch, and an async getter to aquire the data.
+    //     - The getter is a function with signature `(epochRoot, cb)`.
+    //     - e.g.:
+    //       ```
+    //       const getters = {
+    //         author (epochRoot, cb) {
+    //           getRootFeedId(epochRoot.value.author, cb)
+    //         }
+    //       }
+    //       ```
+    // - `cb` *Function* a callback which receives a @tangle/reduce result.
+    //   This is an object which has produced a tangle of connected "nodes"
+    //   (which each "node" is an epoch here), and we can access the graph
+    //   and nodes under `resolve.graph` (which is a @tangle/graph object)
+
+    pull(
+      // collect epochs and decorate each epoch "node" with data (using getters)
+      epochNodeStream(ssb, groupId, { getters }),
+      pull.collect((err, nodes) => {
+        // prettier-ignore
+        if (err) return cb(clarify(err, 'Failure collecting epoch messages'))
+
+        // reduce all of these epoch nodes with a tangle
+        const reduce = new Reduce(strategy, { nodes })
+        reduce.resolve()
+        // INFO: this walks the graph and prunes disconnected nodes
+
+        cb(null, reduce)
+      })
+    )
+  }
+  // emits updated @tangle/reduce objects as changes occur to epoch graph
+  epochsReduce.stream = function epochsReduceStream(groupId, opts = {}) {
+    const { getters, live: isLive } = opts
+    const reduce = new Reduce(strategy)
+
+    return pull(
+      // reduce can be modified by 2 sorts of update:
```

Comment:
The problem was here - `epochsReduceStream` was not emitting new updates when epochs changed in ways that count. Formally the epoch tangle would be emitted ONLY if a new epoch was discovered... and the membership of that epoch would be fixed in the tangle data at that point - regardless of whether new `add-member` messages came in.

This is a problem for calculating `preferredEpoch` - as that consumes this data, and assumes the tangle has all the info it needs.

This updates the epoch tangle graph as new people are added to the graph.
(could have done the same with `exclude-member` but will defer till we need that live updating)

PHEW